src/MultigridMesh.f90

```diff
@@ -0,0 +1,418 @@
+        CALL eLinearSolverType%raiseError(modName//"::"//myName//" - "// &
+          "input MultigridMeshStructure must be initialized!")
+
+      myWtStructure%myMeshes => myMeshes
+
+      ALLOCATE(myWtStructure%wts_level(2:myMeshes%nLevels))
+      DO iLevel=2,myMeshes%nLevels
+        tmp_mesh_ptr => myMeshes%meshes(iLevel)
+        CALL myWtStructure%wts_level(iLevel)%init(tmp_mesh_ptr,num_eqns)
+               !myMeshes%meshes(iLevel),num_eqns)
+      ENDDO
+      NULLIFY(tmp_mesh_ptr)
+
+      myWtStructure%isInit=.TRUE.
+
+    ENDSUBROUTINE init_InterpWeightsStructure
+!
+!-------------------------------------------------------------------------------
+!> @brief Clears the InterpWeightsStructure type
+!>
+!> @param myWtStructure the interp weight structure instance
+!>
+    SUBROUTINE clear_InterpWeightsStructure(myWtStructure)
+      CHARACTER(LEN=*),PARAMETER :: myName='clear_InterpWeightsStructure'
+      CLASS(InterpWeightsStructureType),INTENT(INOUT) :: myWtStructure
+
+      INTEGER(SIK) :: iLevel
+
+      IF(ASSOCIATED(myWtStructure%myMeshes) .AND. &
+          myWtStructure%myMeshes%isInit) THEN
+        IF(ALLOCATED(myWtStructure%wts_level)) THEN
+          DO iLevel=2,myWtStructure%myMeshes%nLevels
+            CALL myWtStructure%wts_level(iLevel)%clear()
+          ENDDO
+          DEALLOCATE(myWtStructure%wts_level)
+        ENDIF
+        NULLIFY(myWtStructure%myMeshes)
+      ELSE IF(ALLOCATED(myWtStructure%wts_level)) THEN
+        DO iLevel=LBOUND(myWtStructure%wts_level,1), &
+                  UBOUND(myWtStructure%wts_level,1)
+          NULLIFY(myWtStructure%wts_level(iLevel)%meshes)
+          CALL myWtStructure%wts_level(iLevel)%clear()
+        ENDDO
+        DEALLOCATE(myWtStructure%wts_level)
+      ENDIF
+
+      myWtStructure%isInit=.FALSE.
+
+    ENDSUBROUTINE clear_InterpWeightsStructure
+!
+!-------------------------------------------------------------------------------
+!> @brief Initializes the InterpWeightsLevel type
+!>
+!> @param myWts the interp weight level instance
+!> @param meshes pointer to MultigridMeshType corresponding to this level
+!> @param num_eqns number of equations (# of unknowns per spatial point)
+!>
+    SUBROUTINE init_InterpWeightsLevel(myWts,meshes,num_eqns)
+      CHARACTER(LEN=*),PARAMETER :: myName='init_InterpWeightsLevel'
+      CLASS(InterpWeightsLevelType),INTENT(INOUT) :: myWts
+      TYPE(MultigridMeshType),POINTER,INTENT(IN) :: meshes
+      INTEGER(SIK),INTENT(IN) :: num_eqns
+
+      INTEGER(SIK) :: i
+
+      myWts%meshes => meshes
+      myWts%num_eqns=num_eqns
+
+      !TODO turn on option for flat weights to save on memory cost
+      !ALLOCATE(myWts%wts_point(meshes%istt:meshes%istp))
+      !DO i=meshes%istt,meshes%istp
+      !  IF(meshes%interpDegrees(i) > 0) THEN
+      !    ALLOCATE(myWts%wts_point(i)%wts(num_eqns,meshes%interpDegrees(i)*2))
+      !    myWts%wts_point(i)%wts=1.0_SRK/(2*meshes%interpDegrees(i))
+      !  ENDIF
+      !ENDDO
```

Comment:
Do we need to save this code?

Reply:
I had intended to try using non-uniform interpolation weights, but found them to be unnecessary for the reactor problems we were trying.  My intention was that it should be easy for someone to try to properly re-implement it should the need arise, so I left it there commented out.  However, I suppose there is a log of it in the commits, so we could just delete it.

Would you guys prefer that I delete it, or should I just leave it there?